docs/source/format/IPC.rst

```diff
@@ -113,60 +120,37 @@ In the file format, there is no requirement that dictionary keys should be
 defined in a ``DictionaryBatch`` before they are used in a ``RecordBatch``, as long
 as the keys are defined somewhere in the file.
 
-RecordBatch body structure
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+RecordBatch messages
+~~~~~~~~~~~~~~~~~~~~
 
-The ``RecordBatch`` metadata contains a depth-first (pre-order) flattened set of
-field metadata and physical memory buffers (some comments from ``Message.fbs``
-have been shortened / removed): ::
+The ``RecordBatch`` metadata contains a depth-first (pre-order)
+flattened set of field metadata and physical memory buffers::
 
     table RecordBatch {
       length: long;
       nodes: [FieldNode];
       buffers: [Buffer];
     }
 
-    struct FieldNode {
-      length: long;
-      null_count: long;
-    }
-
-    struct Buffer {
-      /// The relative offset into the shared memory page where the bytes for this
-      /// buffer starts
-      offset: long;
-
-      /// The absolute length (in bytes) of the memory buffer. The memory is found
-      /// from offset (inclusive) to offset + length (non-inclusive).
-      length: long;
-    }
-
-In the context of a file, the ``page`` is not used, and the ``Buffer`` offsets use
-as a frame of reference the start of the message body. So, while in a general
-IPC setting these offsets may be anyplace in one or more shared memory regions,
-in the file format the offsets start from 0.
-
-The location of a record batch and the size of the metadata block as well as
-the body of buffers is stored in the file footer: ::
-
-    struct Block {
-      offset: long;
-      metaDataLength: int;
-      bodyLength: long;
-    }
-
-The ``metaDataLength`` here includes the metadata length prefix, serialized
-metadata, and any additional padding bytes, and by construction must be a
-multiple of 8 bytes.
-
-Some notes about this
-
-* The ``Block`` offset indicates the starting byte of the record batch.
-* The metadata length includes the flatbuffer size, the record batch metadata
-  flatbuffer, and any padding bytes
-
-Dictionary Batches
-~~~~~~~~~~~~~~~~~~
+* ``RecordBatch``, which contains the flattened ``FieldNode`` and
```

Comment:
Is this a repetition of the previous sentence? Also you're not describing the `length` member.